File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    List<Integer> list = new ArrayList<>();
    
    public int kthSmallest(TreeNode root, int k) {
        // approach: in-order traversal -> array
        
        helper(root);
        
        return list.get(k - 1);
    }
    
    private void helper(TreeNode node) {
        if (node == null) {
            return;
        }
        
        helper(node.left);
        // in-order processing
        list.add(node.val);
        helper(node.right);
    }
}

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        # approach: inorder traversal -> list
        
        nums = []
        
        def helper(node):
            if not node:
                return
            
            helper(node.left)
            nums.append(node.val)
            helper(node.right)
            
        helper(root)
        return nums[k - 1]